### PR TITLE
Cleaning cmake & friends

### PR DESCRIPTION
**Pull request by r-barnes**
* Downgrade required cmake version
* Eliminate extraneous CUDA headers
* Make cross-system compatible dllexport macros

### CODE REVIEW

File: src/Forwarding_DLL.h [hunk 1]
```diff
@@ -1,2 +1,12 @@
...
+#pragma once
+
+#ifdef _MSC_VER
+	#define DLLEXPORT __declspec(dllexport)
+#else
+	#define DLLEXPORT
+#endif
+
+extern "C" {
+	double* DLLEXPORT Forwarding(int lx, int ly, int point_count, int prism_count, double x_min, double dx, double x_max,
```
Comment (AlbertHZL) [inline src/Forwarding_DLL.h]: The revised codes could not be built successfully. We edited here, put "DLLEXPORT" first.

File: src/Forwarding_DLL.h [hunk 2]
```diff
@@ -1,2 +1,12 @@
...
+#pragma once
+
+#ifdef _MSC_VER
+	#define DLLEXPORT __declspec(dllexport)
+#else
+	#define DLLEXPORT
+#endif
+
+extern "C" {
+	DLLEXPORT double* Forwarding(int lx, int ly, int point_count, int prism_count, double x_min, double dx, double x_max,
```
Comment (AlbertHZL) [inline src/Forwarding_DLL.h:10]: The revised could not be built successfully, so we edited here, putting "DLLEXPORT" first.

File: src/grav_rfi_ompcuda.h [hunk 3]
```diff
@@ -1,6 +1,14 @@
...
 
-extern "C"  __declspec(dllexport) double* foo(int h_point_count, int h_prism_count, int h_lx, int h_ly, int h_lz, int h_kmax, int a, int b, double h_z_obs, double h_dz, double h_zmax,
+#ifdef _MSC_VER
+	#define DLLEXPORT __declspec(dllexport)
+#else
+	#define DLLEXPORT
+#endif
+
+extern "C" {
+	DLLEXPORT double* foo(int h_point_count, int h_prism_count, int h_lx, int h_ly, int h_lz, int h_kmax, int a, int b, double h_z_obs, double h_dz, double h_zmax,
```
Comment (AlbertHZL) [inline src/grav_rfi_ompcuda.h:10]: The revised could not be built successfully, so we edited here, putting "DLLEXPORT" first.

File: src/grav_rfi_ompcuda.h [hunk 4]
```diff
@@ -1,6 +1,14 @@
...
 	double h_m_min, double h_m_max, double h_epsilon, double h_lambda, double h_sigma,double wn, double* zc, double* thick, double* Vz, double* x, double* y);
 
-extern "C"  __declspec (dllexport) int CheckCount();
+	DLLEXPORT int CheckCount();
```
Comment (AlbertHZL) [inline src/grav_rfi_ompcuda.h:13]: The revised could not be built successfully, so we edited here, putting "DLLEXPORT" first.

**Review by AlbertHZL — COMMENTED** [review]
The codes with "DLLEXPORT" have been revised.

File: src/Forwarding_DLL.h (hunk 2 above)
Comment (r-barnes) [reply src/Forwarding_DLL.h:10]: lgtm (looks good to me)

File: src/grav_rfi_ompcuda.h (hunk 3 above)
Comment (r-barnes) [reply src/grav_rfi_ompcuda.h:10]: lgtm

File: src/grav_rfi_ompcuda.h (hunk 4 above)
Comment (r-barnes) [reply src/grav_rfi_ompcuda.h:13]: lgtm

**Comment by r-barnes** [comment]
This all looks good to me, thanks!

**Pull request merged**